metrics/expvar/expvar.go

```diff
@@ -75,9 +81,45 @@ type histogram struct {
 	hist *hdrhistogram.WindowedHistogram
 
 	name   string
+	count  metrics.Counter
 	gauges map[int]metrics.Gauge
 }
 
+// Name returns the name of the histogram
+func (h *histogram) Name() string {
+	return h.name
+}
+
+// Print out nonzero bars of the histogram as a csv with normalized probability and a crude bar graph
+func (h histogram) String() string {
+	var total float64
+	d := h.hist.Merge().Distribution()
+	max := 0.0
+	for _, b := range d {
+		c := float64(b.Count)
+		total += c
+		if c > max {
+			max = c
+		}
+	}
+	max = max / total // max probability
+	f := "%8v,%8v,%8v,%7v, %v\n"
+	bs := "####################################################################################################"
+	flbs := float64(len(bs)) - 1.0 // -1.0 to avoid running off the end of the slice index
+	axis := "|"
+	s := fmt.Sprintf(f, "From", "To", "Count", "Prob", "Bar")
+	for _, b := range d {
+		if b.Count > 0 {
+			p := float64(b.Count) / total
+			s += fmt.Sprintf(f, b.From, b.To, b.Count, fmt.Sprintf("%0.4f", p), axis+bs[:int(p*flbs/max)])
+			axis = "|"
+		} else {
+			axis = ":" // show that some bars were skipped
+		}
+	}
+	return fmt.Sprintf("name: %v\ncount: %v\ngauges: %v\n%v\n", h.name, h.count, h.gauges, s)
+}
+
```

Comment:
Here's a strawman implementation using text.TabWriter. Adjust as you see fit.

```go
const (
	bs   = "####################################################################################################"
	flbs = float64(len(bs))
)

func (h *histogram) String() string {
	d := h.hist.Merge().Distribution()

	var total float64
	for _, b := range d {
		total += float64(b.Count)
	}

	var buf bytes.Buffer
	w := tabwriter.NewWriter(&buf, 0, 2, 2, ' ', 0)
	fmt.Fprintf(w, "From\tTo\tCount\tProb\tBar\n")

	axis := "|"
	for _, b := range d {
		if b.Count > 0 {
			p := float64(b.Count) / total
			fmt.Fprintf(w, "%d\t%d\t%d\t%.4f\t%s%s\n", b.From, b.To, b.Count, p, axis, bs[:int(p*flbs)])
			axis = "|"
		} else {
			axis = ":" // show that some bars were skipped
		}
	}

	fmt.Fprintf(&buf, "name: %v\n", h.name)
	fmt.Fprintf(&buf, "count: %v\n", h.count)
	fmt.Fprintf(&buf, "gauges: %v\n", h.gauges)
	w.Flush() // to buf
	return buf.String()
}
```

Sample output:

```
name: test_histogram_ascii
count: <nil>
gauges: map[50:0xc8200d8040 90:0xc8200d8048 95:0xc8200d8050 99:0xc8200d8058]
From  To  Count  Prob    Bar
1     1   2      0.0016  :
2     2   36     0.0292  |##
3     3   157    0.1272  |############
4     4   397    0.3217  |################################
5     5   418    0.3387  |#################################
6     6   188    0.1524  |###############
7     7   34     0.0276  |##
8     8   2      0.0016  |
```